test/gtest/core/test_bcast.cc

```diff
@@ -0,0 +1,134 @@
+                              ctxs[r]->rbuf_size, mem_type,
+                              UCC_MEMORY_TYPE_HOST);
+            }
+        }
+        return ctxs;
+    }
+    void data_fini(UccCollCtxVec ctxs)
+    {
+        for (gtest_ucc_coll_ctx_t* ctx : ctxs) {
+            ucc_coll_args_t* coll = ctx->args;
+            UCC_CHECK(ucc_mc_free(coll->src.info.buffer, mem_type));
+            UCC_CHECK(ucc_mc_free(ctx->init_buf, UCC_MEMORY_TYPE_HOST));
+            free(coll);
+            free(ctx);
+        }
+        ctxs.clear();
+    }
+    void data_validate(UccCollCtxVec ctxs)
+    {
+        int root = ctxs[0]->args->root;
+        uint8_t *dsts;
+
+        if (UCC_MEMORY_TYPE_HOST != mem_type) {
+                UCC_CHECK(ucc_mc_alloc((void**)&dsts, ctxs[root]->rbuf_size,
+                                       UCC_MEMORY_TYPE_HOST));
+                ucc_mc_memcpy(dsts, ctxs[root]->args->src.info.buffer,
+                              ctxs[root]->rbuf_size, UCC_MEMORY_TYPE_HOST, mem_type);
+        } else {
+            dsts = (uint8_t*)ctxs[root]->args->src.info.buffer;
+        }
+        for (int r = 0; r < ctxs.size(); r++) {
+            ucc_coll_args_t* coll = ctxs[r]->args;
+            if (coll->root == r) {
+                continue;
+            }
+            for (int i = 0; i < ctxs[r]->rbuf_size; i++) {
+                EXPECT_EQ(i, dsts[i]);
+            }
+        }
+        if (UCC_MEMORY_TYPE_HOST != mem_type) {
+            ucc_mc_free((void*)dsts, UCC_MEMORY_TYPE_HOST);
+        }
+        return;
+    }
+    void set_root(int _root)
+    {
+        root = _root;
+    }
+};
+
+class test_bcast_0 : public test_bcast,
+        public ::testing::WithParamInterface<Param> {};
+
+UCC_TEST_P(test_bcast_0, single_host)
+{
+    const int                 team_id  = std::get<0>(GetParam());
+    const ucc_datatype_t      dtype    = (ucc_datatype_t)std::get<1>(GetParam());
+    const ucc_memory_type_t   mem_type = std::get<2>(GetParam());
+    const int                 count    = std::get<3>(GetParam());
+    const int                 root     = std::get<4>(GetParam());
+    UccTeam_h                 team     = UccJob::getStaticTeams()[team_id];
+    int                       size     = team->procs.size();
+
+    set_mem_type(mem_type);
+    set_root(root);
+
+    UccCollCtxVec args = data_init(size, (ucc_datatype_t)dtype, count);
+    UccReq    req(team, args);
+    req.start();
+    req.wait();
+    data_validate(args);
+    data_fini(args);
+}
+
+INSTANTIATE_TEST_CASE_P(
+    ,
+    test_bcast_0,
+    ::testing::Combine(
+        ::testing::Range(1, UccJob::nStaticTeams), // team_ids
+        ::testing::Range((int)UCC_DT_INT8, (int)UCC_DT_UINT32 + 1), // dtype
+#ifdef HAVE_CUDA
+        ::testing::Values(UCC_MEMORY_TYPE_HOST, UCC_MEMORY_TYPE_CUDA), // mem type
+#else
+        ::testing::Values(UCC_MEMORY_TYPE_HOST),
+#endif
+        ::testing::Values(1,3), // count
```

Comment:
same here, use some Large count as well (for bcast i would add count = 65536)

Reply:
fixed